src/dash-table/derived/cell/wrapperStyles.ts

```diff
@@ -2,70 +2,56 @@ import * as R from 'ramda';
 import { CSSProperties } from 'react';
 
 import { memoizeOneFactory } from 'core/memoizer';
-import { Data, VisibleColumns, IViewportOffset, SelectedCells, ICellCoordinates } from 'dash-table/components/Table/props';
+import { Data, VisibleColumns, IViewportOffset, SelectedCells } from 'dash-table/components/Table/props';
 import { IConvertedStyle } from '../style';
-import { BORDER_PROPERTIES_AND_FRAGMENTS } from '../edges/type';
+import { getDataCellStyle, getDataOpCellStyle } from '../edges';
+import { traverseMap2, shallowClone } from 'core/math/matrixZipMap';
 
-type Style = CSSProperties | undefined;
+const SELECTED_CELL_STYLE = { backgroundColor: 'var(--selected-background)' };
 
-function getter(
+const partialGetter = (
     columns: VisibleColumns,
-    columnStyles: IConvertedStyle[],
+    styles: IConvertedStyle[],
     data: Data,
+    offset: IViewportOffset
+) => traverseMap2(
+    data,
+    columns,
+    (datum, column, i) => getDataCellStyle(datum, i + offset.rows, column)(styles)
+);
+
+const getter = (
+    styles: CSSProperties[][],
     offset: IViewportOffset,
     selectedCells: SelectedCells
-): Style[][] {
-    return R.addIndex<any, Style[]>(R.map)((datum, index) => R.addIndex<any, Style>(R.map)((column, columnIndex) => {
-        const relevantStyles = R.map(
-            s => s.style,
-            R.filter<IConvertedStyle>(
-                style =>
-                    style.matchesColumn(column) &&
-                    style.matchesRow(index + offset.rows) &&
-                    style.matchesFilter(datum),
-                columnStyles
-            )
-        );
-        const matchCell = (cell: ICellCoordinates) => cell.row === index && cell.column === columnIndex;
-        const isSelectedCell: boolean = R.any(matchCell)(selectedCells);
-        if (isSelectedCell) {
-            relevantStyles.push({backgroundColor:  'var(--selected-background)'});
+) => {
+    styles = shallowClone(styles);
+
+    R.forEach(({ row: i, column: j }) => {
+        i -= offset.rows;
+        j -= offset.columns;
+
+        if (i < 0 || j < 0 || styles.length <= i || styles[i].length <= j) {
```

Comment:
Now looping on `selected_cells` instead of data/column. Need to check boundaries.